src/foam/core/ImportsExports.js

```diff
@@ -163,34 +163,45 @@ foam.CLASS({
   ],
 
   methods: [
+    function getExportMap() {
+      var m = {};
+      var bs = this.cls_.getAxiomsByClass(foam.core.Export);
+      for ( var i = 0 ; i < bs.length ; i++ ) {
+        var b = bs[i];
+
+        if ( b.key ) {
+          var path = b.key.split('.');
+
+          var a = this.cls_.getAxiomByName(path[0]);
+
+          foam.assert(!!a, 'Unknown axiom: "', path[0], '" in model: ',
+                      this.cls_.id, ", trying to export: '", b.key, "'");
+
+          // Axioms have an option of wrapping a value for export.
+          // This could be used to bind a method to 'this', for example.
+          var e = a.exportAs ? a.exportAs(this, path.slice(1)) : this[path[0]];
```

Comment:
No.

Consider that in both cases of exporting `foo` and `foo.bar` we call `exportAs` on the `foo` axiom.  So why would this become inconsistent?

The code here isn't responsible for making sure both foo and foo$ are available in the context.  That's done in createSubContext.  All we do here is let the axiom decide what it wants to export itself as and choose the key we want to export with.